core/arch/arm/plat-stm32mp1/shared_resources.c

```diff
@@ -683,21 +684,26 @@ static void check_rcc_secure_configuration(void)
 		    id == STM32MP1_SHRES_SRAM4)
 			continue;
 
-		if ((mckprot_resource(id) && !mckprot) || !secure) {
-			EMSG("RCC %s MCKPROT %s and %s (%u) secure",
-			      secure ? "secure" : "non-secure",
-			      mckprot ? "set" : "not set",
-			      shres2str_id(id), id);
-			have_error = true;
-		}
+		need_secure = true;
+		if (mckprot_resource(id))
+			need_mckprot = true;
+
+		if (!secure || (need_mckprot && !mckprot))
+			EMSG("Error RCC TZEN=%u MCKPROT=%u and %s (%u) secure",
+			     secure, mckprot, shres2str_id(id), id);
 	}
 
-	if (have_error) {
+	if ((need_secure && !secure) || (need_mckprot && !mckprot)) {
 		if (IS_ENABLED(CFG_INSECURE))
 			EMSG("WARNING: CFG_INSECURE allows insecure RCC configuration");
 		else
 			panic();
 	}
+
+	if (!need_mckprot && mckprot) {
```

Comment:
This modifies the conf in a "check" function. I'd either put this in the caller function `stm32mp1_init_final_shres` or rename the function.

Otherwise, the content itself looks good

Reply:
Ok, i'll rename the function to `rcc_secure_configuration()`